libs/blocks/merch-card/merch-card.js

```diff
@@ -168,17 +162,60 @@ const init = (el) => {
       }
     }
   });
-  addWrapper(el, section, cardType);
-  merchCard.classList.add('consonant-Card', 'consonant-ProductCard', `consonant-${cardType}`);
-  if (image) addBackgroundImg(image, cardType, merchCard);
-  if (ribbonMetadata !== null) decorateRibbon(el, ribbonMetadata, cardType);
-  image?.parentElement.remove();
-  if (ctas) decorateButtons(ctas);
-  addInner(el, altCta, cardType, merchCard);
-  decorateIcon(el, icons, cardType);
-  const inner = el.querySelector(`.consonant-${cardType}-inner`);
-  const innerCleanup = inner.querySelectorAll(':scope > div')[1];
-  if (innerCleanup.classList.length === 0) innerCleanup.remove();
+
+  const merchCard = createTag('merch-card', { class: el.className, variant: cardType });
+  if (ribbonMetadata !== null) {
+    const badge = returnRibbonStyle(ribbonMetadata);
+    if (badge !== null) {
+      merchCard.setAttribute('badge-background-color', badge.badgeBackgroundColor);
+      merchCard.setAttribute('badge-color', badge.badgeColor);
+      merchCard.setAttribute('badge-text', badge.badgeText);
+    }
```

Comment:
Looks like you can un nest

```suggestion
  const badge = returnRibbonStyle(ribbonMetadata);
  if (badge !== null) {
    merchCard.setAttribute('badge-background-color', badge.badgeBackgroundColor);
    merchCard.setAttribute('badge-color', badge.badgeColor);
    merchCard.setAttribute('badge-text', badge.badgeText);
  }
```